packages/features/sales-escrows/pages/NFTPage/index.tsx

```diff
@@ -167,17 +167,30 @@ export const NFTPage = () => {
 
         if ('err' in r) throw new Error(Object.keys(r.err)[0]);
 
-        const dataObj = r?.ok.metadata.Class.find(({ name }) => name === '__apps')
-          .value.Array.thawed[0].Class.find(({ name }) => name === 'data')
-          .value.Class.reduce(
-            (arr, val) => ({ ...arr, [val.name]: Object.values(val.value)[0] }),
-            {},
-          );
-        dataObj.tokenID = r?.ok?.metadata?.Class?.find(({ name }) => name === 'id').value.Text;
+        const dataObj: { [key: string]: string } = {
+          tokenID: r?.ok?.metadata?.Class?.find(({ name }) => name === 'id').value.Text || '',
+        };
+
+        const dataObj2 = r?.ok?.metadata?.Class?.find(
+          ({ name }) => name === '__apps',
+        )?.value?.Array?.thawed[0]?.Class?.find(({ name }) => name === 'data')?.value?.Class || '';
+
+        if (dataObj2) {
+          dataObj2.map((item) => {
+            dataObj[item.name] = item.value.Text || '';
+            if (item.name == 'custom_properties') {
+              item.value.Array.thawed.map((item) => {
+                dataObj[item.name] = item.value.Text || '';
```

Comment:
Why is this reassigning dataObj[item.name] to the last custom_property text value?
Why use the map function for this?
Are we sure that each custom property is a text value?